### PR TITLE
Basic http auth snippet

### PR DESCRIPTION
**Pull request by maximivanov**
Adding an example with Basic HTTP authentication as suggested by @victoriabernard92 

Source PR: https://github.com/south-lane/cf-worker-basic-http-auth/pull/1

### CODE REVIEW

**Comment by maximivanov** [comment]
I may need to reorder things in code to follow the guideline. On it.

**Review by exvuma — COMMENTED** [review]
Thanks for adding this could you make sure to add an [id.toml](https://github.com/ashleygwilliams/template-registry/blob/master/CONTRIBUTING.md#rust-styleguide) ?

**Comment by maximivanov** [comment]
@victoriabernard92 toml file added. I was not sure about `share_url` as there's some difference in docs and in other snippets on how to define it. Some do not even have it so I guess it's no longer used.

Also reordered code blocks. Let me know if it works.

**Comment by maximivanov** [comment]
@victoriabernard92 anything else to fix in this PR?

File: templates/meta_data/auth_basic_http.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+id = "auth_basic_http"
+weight = 1
+type = "snippet"
+title = "Basic HTTP authentication"
+description = "Password-protect whole websites or specific areas/pages. E.g. limit access to a staging version of a website you're building."
+share_url="/templates/pages/auth_basic_http"
```
Comment (exvuma) [inline templates/meta_data/auth_basic_http.toml]:
```suggestion
```

You're right you don't actually need this

File: templates/meta_data/auth_basic_http.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+id = "auth_basic_http"
+weight = 1
+type = "snippet"
+title = "Basic HTTP authentication"
+description = "Password-protect whole websites or specific areas/pages. E.g. limit access to a staging version of a website you're building."
```
Comment (exvuma) [inline templates/meta_data/auth_basic_http.toml]:
```suggestion
description = "Password-protect whole websites or specific areas/pages. Browsers will prompt for a password and username - hardcoded as `demouser`/`demopassword` - before serving a protected page. This uses basic authentication through the [`WWW-Authorization` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication).
```

**Review by exvuma — CHANGES_REQUESTED** [review]

**Comment by exvuma** [comment]
Thanks @maximivanov almost there let me know what you think of my suggestions then I think we can merge

**Comment by maximivanov** [comment]
Hey @victoriabernard92 thanks for the suggestions, they look great!

Not sure why the build is failing though. It seems like it can't find `./dist/index.js`... does it look in project's root while `dist` folder is in `workers-site/`?

I git-rebased latest changes from the upstream repo's `master` to my branch but it didn't help.

**Comment by exvuma** [comment]
@maximivanov Builds are broken because CI isn't running the build step. Not this PR's fault I opened an issue because I've been meaning to fix this:https://github.com/ashleygwilliams/template-registry/issues/23

Sorry for the trouble there.

**Review by exvuma — APPROVED** [review]

**Comment by maximivanov** [comment]
Hi @victoriabernard92 I just realized this snippet is not appearing on the templates site... not sure whether it's expected or if something is broken.

**Pull request merged**